internal/api/update_credentials_v1alpha1.go

```diff
@@ -80,7 +78,29 @@ func (s *server) UpdateCredentials(
 		)
 	}
 
-	applyCredentialsUpdateToSecret(&secret, credsUpdate)
+	switch credType {
```

Comment:
This switch can be improved in the same way described [here](https://github.com/akuity/kargo/pull/2975/files#r1910720424)